module/data/abstract.mjs

```diff
@@ -56,7 +56,23 @@ export default class SystemDataModel extends foundry.abstract.DataModel {
    * @returns {object}  Fully merged schema.
    */
   static mergeSchema(a, b) {
-    Object.assign(a, b);
+    for ( const key of Object.keys(b) ) {
+      if ( !(key in a) || (a[key].constructor !== b[key].constructor) ) {
+        a[key] = b[key];
+        continue;
+      }
+      switch (b[key].constructor) {
+        case foundry.data.fields.SchemaField:
+          const fields = this.mergeSchema(a[key].fields, b[key].fields);
+          Object.values(fields).forEach(f => f.parent = undefined);
+          a[key] = new foundry.data.fields.SchemaField(
+            fields, foundry.utils.mergeObject(a[key].options, b[key].options)
+          );
+          break;
+        default:
+          a[key] = b[key];
```

Comment:
Perhaps some solution for merging the `options` of the two fields at least here.

Reply:
I have a solution for that over on my actor data branch where we actually would need it, but I figured I would only include the code here we are actually using.